src/ImageSharp/Common/Extensions/StreamExtensions.cs

```diff
@@ -82,5 +83,22 @@ public static void Write(this Stream stream, IManagedByteBuffer buffer)
         {
             stream.Write(buffer.Array, 0, buffer.Length());
         }
+
+#if NET472 || NETSTANDARD1_3 || NETSTANDARD2_0
+        // This is a port of the CoreFX implementation and is MIT Licensed: https://github.com/dotnet/coreclr/blob/c4dca1072d15bdda64c754ad1ea474b1580fa554/src/System.Private.CoreLib/shared/System/IO/Stream.cs#L770
+        public static void Write(this Stream stream, ReadOnlySpan<byte> buffer)
+        {
+            byte[] sharedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
+            try
+            {
+                buffer.CopyTo(sharedBuffer);
```

Comment:
To me it looks like we are loosing all the benefits of the copy-free initialization at this line...

Reply:
But the benchmarks show different results ... I wonder why?

Reply:
In the case of netcoreapp, the method is actually virtual and various stream types can override it to avoid the copy. I'll double check the numbers on full framework to see if it hurts anything (and if so, will look at what can be done).